net_utils.go

```diff
@@ -207,6 +207,41 @@ func incrementIP(ip net.IP) (result net.IP) {
 	return
 }
 
+// decrementIP returns the given IP 1 1
+func decrementIP(ip net.IP) (result net.IP) {
+	// copy the ip into the result
+	result = make([]byte, len(ip))
```

Comment:
I do like the use of `make` here better than what I did in incrementIP.

Reply:
I found that somewhere on the web :)